c10/cuda/CUDAStream.cpp

```diff
@@ -260,10 +214,9 @@ static void initCUDAStreamsOnce() {
   }
 
   // Inits current streams (thread local) to default streams
-  current_streams =
-      (LeakyStreamInternals**)malloc(num_gpus * sizeof(LeakyStreamInternals*));
```

Comment:
I'm not an expert with thread-local memory, but doesn't this cause a memory leak? If a new thread is spawned, and it sets a current stream, and then terminates, doesn't this allocation remain around?

Perhaps we could fix it by using a `std::vector` here (assuming the destructors of thread-local objects are invoked when a thread terminates). Or alternatively use a fixed-size array instead of doing a dynamic heap allocation.